src/components/combo/combo.ts

```diff
@@ -0,0 +1,654 @@
+
+  /**
+   * Enables the case sensitive search icon in the filtering input.
+   * @attr case-sensitive-icon
+   */
+  @property({ type: Boolean, attribute: 'case-sensitive-icon', reflect: false })
+  public caseSensitiveIcon = false;
+
+  /**
+   * Disables the filtering of the list of options.
+   * @attr disable-filtering
+   */
+  @property({ type: Boolean, attribute: 'disable-filtering', reflect: false })
+  public disableFiltering = false;
+
+  /**
+   * The template used for the content of each combo item.
+   * @type {(item: T) => TemplateResult}
+   */
+  @property({ attribute: false })
+  public itemTemplate: (item: ComboRecord<T>) => TemplateResult = (item) => {
+    if (typeof item !== 'object' || item === null) {
+      return String(item) as any;
+    }
+
+    if (this.displayKey) {
+      return html`${item[this.displayKey]}`;
+    }
+
+    return html`${String(item)}`;
+  };
+
+  /**
+   * The template used for the content of each combo group header.
+   * @type {(item: T) => TemplateResult}
+   */
+  @property({ attribute: false })
+  public groupHeaderTemplate: (item: ComboRecord<T>) => TemplateResult = (
+    item: ComboRecord<T>
+  ) => {
+    return html`${this.groupKey && item[this.groupKey]}`;
+  };
+
+  /** @hidden @internal */
+  @state()
+  public dataState: Array<ComboRecord<T>> = [];
+
+  @watch('data')
+  protected dataChanged() {
+    this.dataState = structuredClone(this.data) as ComboRecord<T>[];
+
+    if (this.hasUpdated) {
+      this.pipeline();
+    }
+  }
+
+  @watch('valueKey')
+  protected updateDisplayKey() {
+    this.displayKey = this.displayKey ?? this.valueKey;
+  }
+
+  @watch('displayKey')
+  protected updateFilterKey() {
+    this.filteringOptions.filterKey =
+      this.filteringOptions.filterKey ?? this.displayKey;
+  }
+
+  @watch('groupKey')
+  @watch('groupSorting')
+  @watch('pipeline')
+  protected async pipeline() {
+    this.dataState = await this.dataController.apply([...this.data]);
+    this.navigationController.active = 0;
+  }
+
+  @watch('open')
+  protected toggleDirectiveChange() {
+    if (!this.target) return;
+    this.toggleController.target = this.target;
+    this.target.setAttribute('aria-expanded', this.open ? 'true' : 'false');
+  }
+
+  constructor() {
+    super();
+
+    this.toggleController = new IgcToggleController(this, {
+      target: this.target,
+      closeCallback: async () => {
+        if (!this.handleClosing()) return;
+        this.open = false;
+
+        await this.updateComplete;
+        this.emitEvent('igcClosed');
+      },
+    });
+
+    this.addEventListener('focus', () => {
+      this.emitEvent('igcFocus');
+    });
+
+    this.addEventListener('blur', () => {
+      this.emitEvent('igcBlur');
+    });
+
+    this.addEventListener(
+      'keydown',
+      this.navigationController.navigateHost.bind(this.navigationController)
+    );
+  }
+
+  protected override async firstUpdated() {
+    await this.updateComplete;
+    this.requestUpdate();
```

Comment:
The combo requests and update for what exactly?

Reply:
By *encapsulated input* you mean the actual input of the combo and the filtering one, right?
Are they not correctly rendered initially? At least in the storybook it seems to work without the additional call?

Reply:
It didn't work initially. I will check again.

Reply:
You are right, it works without it. Maybe something else is triggering it now as it didn't work initially. Anywho, I've removed it.